### ISSUE TITLE
NoLoggingT for Katip

### ISSUE THREAD

**Issue by bitemyapp** [Soostone/katip#26]
[NoLoggingT from monad-logger](https://hackage.haskell.org/package/monad-logger-0.3.18/docs/Control-Monad-Logger.html#t:NoLoggingT)

Currently trying to figure out how to make a null logger for Katip. If this has bin-dun-befo then I couldn't find it.

**Comment by MichaelXavier** [comment]
Just to clarify, would this be a transformer you introduce that would cease logging to any of the existing scribes? I hesitate to go down the path of using transformer stack layers for features only because they get messy pretty quickly. You have to rederive all of the supported instances for this stack and if the user has any orphaned instances they have to do the same.

If I were to do this in my own application, I'd do it as a combinator, kind of like [katipAddNamespace](https://github.com/Soostone/katip/blob/master/katip/src/Katip/Monadic.hs#L314). Most use cases will involve a ReaderT over LogEnv at some level, so dropping all handles would be really simple, something like `local clearScribes`. In fact, I think it would be reasonable to add a `katipPauseLogging` or something that did this very thing.

The guiding principle I'm working with right now is to provide some basic reference implementations (`KatipContextT`, analogous to the reference implementation `ReaderT` that implements `MonadReader`) and have that come with good examples of common operations like we're discussing here, which can be easily implemented if the user uses their own stack.

I'm totally open to being convinced about doing this with a transformer layer, but I prefer the ergonomics of the `mtl` where you usually work just with constraints and only once construct your concrete transformer stack.

**Comment by bitemyapp** [comment]
@MichaelXavier I'd rather it be done your way, I'm not really proposing `NoLoggingT`'s ergonomics precisely so much as calling it out as a potential need/use-case to be provisioned for, by a combinator or otherwise. I'm trying not to change Katip too much while I'm still integrating and learning it for our projects at work.

Most of my work of late has been ripping out the `monad-logger` that proliferated because I was doing a pretty vanilla @snoyberg stack.

**Comment by MichaelXavier** [comment]
Gotcha. I think I'll move forward with adding that combinator then and even a case for the example files. I think this is a very common need.

I wonder if you could actually ease the transition by defining an instance of `MonadLogger` that calls the appropriate katip function. All the arguments to `monadLoggerLog ` seem analogous to arguments in Katip. Maybe a migration guide could exist in the readme and example code. I can't really justify providing any direct migration code in katip as it would add a `monad-logger` dependency which seems inappropriate. I could see adding a `katip-monad-logger-compat` package or something if there were enough demand though.

**Comment by MichaelXavier** [comment]
Take a look at the above commit and let me know what you think.

Regarding MonadLogger compat, let me know if you think any of my suggestions are worth pursuing and if so, I'll create another issue.

**Comment by MichaelXavier** [comment]
@bitemyapp last call before I merge this. It seems a pretty harmless change but considering you may be the first user, I want to make sure it satisfies the need.

**Comment by bitemyapp** [comment]
@MichaelXavier that looks like it'll work, thank you!
